camp/asher-miniproject/src/main/java/frc/robot/subsystems/MotorSubsystem.java

```diff
@@ -0,0 +1,147 @@
+package frc.robot.subsystems;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.StatusCode;
+import com.ctre.phoenix6.configs.CANcoderConfiguration;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import static java.lang.Math.max;
+import static java.lang.Math.min;
+
+import frc.robot.Constants;
+
+/**
+ * The MotorSubsystem is responsible for controlling a motor and its associated encoder.
+ * It provides methods to set motor speed, retrieve the position of the encoder, and configure
+ * the motor and encoder settings.
+ *
+ * <p>This subsystem uses a TalonFX motor controller and a CANcoder encoder. The motor speed
+ * is clamped to a range of -1 to 1, and the encoder position is returned in degrees.
+ *
+ * <p>Key features:
+ * - Configures motor settings such as neutral mode, voltage limits, and current limits.
+ * - Configures encoder settings such as magnet offset and sensor direction.
+ * - Provides utility methods for clamping values and wrapping degrees.
+ */
+public class MotorSubsystem extends SubsystemBase{
+    private final TalonFX motor;
+        private final DutyCycleOut eeDutyCycle = new DutyCycleOut(0.0);
+
+    private final CANcoder motorEncoder;
+
+
+    /**
+     * Constructs a MotorSubsystem with a motor and an encoder.
+     *
+     * @param motorCan The CAN ID of the motor controller (TalonFX).
+     * @param encoderCan The CAN ID of the encoder (CANcoder).
+     */
+    public MotorSubsystem(int motorCan, int encoderCan){
+        this.motor = new TalonFX(motorCan);
+        configureMotor();
+
+        this.motorEncoder = new CANcoder(encoderCan);
+        configureAngleEncoder();
+    }
+
+    /**
+     * Sets the speed of the motor within the allowed range of -1 to 1.
+     *
+     * @param motorSpeed The desired speed of the motor. Values are clamped between -1 (full reverse) and 1 (full forward).
+     */
+    public void setMotorSpeed(double motorSpeed){
+        eeDutyCycle.withOutput(clamp(motorSpeed, -1, 1));
```

Comment:
we want to now use edu.wpi.first.math.MathUtil.clamp instead of writing our own.